### ISSUE TITLE
numerical issue in TruncatedNormal::computeScalarQuantile

### ISSUE THREAD

**Issue by jschueller** [openturns/openturns#1546]
When the normalization factor is too high (ie when the truncation interval contains a very low probability)
and when the lower bound is far (PhiANorm_=1)
computeScalarQuantile is not well conditioned:
```
Scalar TruncatedNormal::computeScalarQuantile(const Scalar prob) const
{
  return mu_ + sigma_ * DistFunc::qNormal(PhiANorm_ + prob / normalizationFactor_);
}
```

```
import openturns as ot
dist = ot.TruncatedNormal(-52.0658, 1.56137, -2.22475e-05, 0.1408)
print([dist.computeScalarQuantile(0.1*i) for i in range(10)])
[-39.37827809285439, -39.37827809285439, -39.37827809285439, -39.37827809285439, -39.37827809285439]
```
I guess TruncatedNormal should not accept such a low probability, but it's not clear yet where to cap things.

On the another side, TruncatedDistribution is safer as it raises an exception in the ctor because it says the probability of the interval is not >0
```
ot.TruncatedDistribution(ot.Normal(-52.0658, 1.56137), -2.22475e-05, 0.1408)
InvalidArgumentException : Error: the truncation interval does not contain a non-empty part of the support of the distribution
```
Actually that's not true:
```
print(ot.Normal(-52.0658, 1.56137).computeProbability(ot.Interval([-2.22475e-05], [0.1408]))
3.930127896167556e-244
```

The thing is TruncatedDistribution computes first the intersection of the range and the bounds:
```
interval1 = ot.Normal(-52.0658, 1.56137).getRange()
interval2 = ot.Interval([-2.22475e-05], [0.1408])
print(interval1, interval2, interval1.intersect(interval2))
]-inf (-64.0113), (-40.1203) +inf[ [-2.22475e-05, 0.1408] [-2.22475e-05, -40.1203]
```
So the resulting interval is `[-2.22475e-05, -40.1203]` and computeProbability returns 0 (because a>b)

I'm wondering if we should check a>b in Interval methods

**Comment by regislebrun** [comment]
A few comments here:

- Don't rely on OT for very small values of probabilities such as `3.930127896167556e-244` resulting from `computeProbabilities()` as it may be the result of a difference with a lot of cancellation
- If you allow a>b in intervals, you open the door to many awful bugs, in particular in all the distributions relying on the default algorithms implemented in `DistributionImplementation` and based on numerical integration.
- In the case of the Normal distribution, we could derive specific approximations for extreme truncation. For example, in your case a truncated Exponential approximation should work if I remember well. I mean: the PDF of your truncated Normal distribution can be written as the PDF of an Exponential distribution times a converging series.

**Comment by jschueller** [comment]
@regislebrun, look at the the result of interval1.intersect(interval2): it has a>b

**Comment by regislebrun** [comment]
@jschueller You are right, I had understood that you wanted to remove a test! I checked the code in `Interval` to remind me what was the intent of the numerical aspects and the mathematical aspects, and found a bug in `Interval::isEmpty`. It is correct if both bounds are finite, but there are 3 other cases to consider (assuming a>b):
- if both bounds are infinite, if a and b have the same sign, the interval is either `]-inf, -inf[` or `]inf, inf[` and the interval is empty, and if they are of opposite sign, b<0 and a>0 and the interval is `]inf, -inf[` and is empty. The method will return False in this case, which is a bug.
- if a is finite and not b, then if b<0 the interval is `[a, -inf[` so it is empty. The method will return False in this case, which is a bug. Otherwise the interval is `[a, +inf[` and it is not empty even if numerically it should be seen as the empty interval
- if b is finite and not a, then if a>0 the interval is `]+inf, b]` so it is empty. The method will return False in this case, which is a bug. Otherwise the interval is `]-inf, b]` and it is not empty even if numerically it should be seen as the empty interval

**Comment by jschueller** [comment]
so `[-2.22475e-05, -40.1203]` should be considered as an empty interval, and computeprobability is actually correct to return 0

**Comment by regislebrun** [comment]
So what do you propose for an empty interval? a<=b?

**Comment by jschueller** [comment]
a=b works too

**Comment by regislebrun** [comment]
Not really, as a\in[a,a]. Think about a discrete distribution.

**Comment by regislebrun** [comment]
If the test x\in[a,b] is done using x>=a and x<=b, then [a,b] is empty if and only if the test always fail if and only if a>b

**Comment by jschueller** [comment]
ok
